sdks/python/apache_beam/ml/inference/sklearn_inference.py

```diff
@@ -156,7 +174,8 @@ class SklearnModelHandlerPandas(ModelHandler[pandas.DataFrame,
   def __init__(
       self,
       model_uri: str,
-      model_file_type: ModelFileType = ModelFileType.PICKLE):
+      model_file_type: ModelFileType = ModelFileType.PICKLE,
+      inference_fn: Optional[Callable[[BaseEstimator, Sequence[pandas.DataFrame], Optional[Dict[str, Any]]], Any]] = _default_pandas_inference_fn):
```

Comment:
In addition to Optional, do we want to make this keyword-only argument? Just putting a restriction on the Optional Params

https://peps.python.org/pep-3102/

Reply:
That's an interesting idea. Stylistically is this something we'd want to adopt in the Python SDK?

Reply:
Hm..I haven't seen it in the codebase. it's a PEP proposed feature https://peps.python.org/pep-3102/ so maybe we can use it? @tvalentyn what do you think?

Reply:
It's a Python 3.0 feature so we could absolutely use it, it's purely a style consideration at this point

Reply:
it is useful to make sure people don't use positional arguments if doing so would be error-prone (for example for functions taking many args, or if you are changing the order of the existing args).

Reply:
What are we doing here? if we add this later, this could be a breaking change for user who is already using it since numpy Modelhandler is backwards compatible.

I am +1 to add this.

Reply:
I can drop in the single-star character into the constructor and see how it looks. Shouldn't be a big change and helps make things explicit.

Reply:
Done.

Reply:
This is neat, I like where it landed